### PR TITLE
[GLUTEN-7240][CH] Fix all failed uts in GlutenComplexTypeSuite

### PR DESCRIPTION
**Pull request by taiyang-li**
## What changes were proposed in this pull request?
Fix GlutenComplexTypeSuite. Close  \#7240

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: gluten-ut/spark33/src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala [hunk 1]
```diff
@@ -692,11 +692,6 @@ class ClickHouseTestSettings extends BackendTestSettings {
       "SPARK-36740: ArrayMin/ArrayMax/SortArray should handle NaN greater then non-NaN value")
     .excludeGlutenTest("Shuffle")
   enableSuite[GlutenComplexTypeSuite]
-    .exclude("SPARK-33386: GetArrayItem ArrayIndexOutOfBoundsException")
```
Comment (baibaichen) [inline gluten-ut/spark33/src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala:695]: how about spark 32 and spark 35 ?

Comment (taiyang-li) [reply gluten-ut/spark33/src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala:695]: It could also be removed in spark 32 and 35

Comment (taiyang-li) [reply gluten-ut/spark33/src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala:695]: done

**Review by baibaichen — APPROVED** [review]
LGTM

**Pull request merged**